core/src/Kokkos_MathematicalFunctions.hpp

```diff
@@ -91,8 +91,12 @@ namespace Experimental {
 #if defined(KOKKOS_ENABLE_SYCL)
 #define KOKKOS_IMPL_MATH_FUNCTIONS_NAMESPACE sycl
 #else
+#if defined(KOKKOS_ENABLE_CUDA) && defined(__GNUC__) && (__GNUC__ < 6)
```

Comment:
Is it a problem that this path might also be taken by a compiler different from `g++`, like `icpc` or `clang++`?

Reply:
what do you mean? The issue is the C++ cmath header, which should only depend on the GNUC

Reply:
or are you worried this is getting taken by clang when it is supposed to use its own headers?

Reply:
`clang` doesn't like this:
```
/var/jenkins/workspace/Kokkos/core/src/Kokkos_MathematicalFunctions.hpp:298:34: error: reference to __device__ function 'isinf' in __host__ __device__ function [clang-diagnostic-error]
```